packages/engine/src/parseErrors.ts

```diff
@@ -1,39 +1,66 @@
 import tsErrorMessages from './tsErrorMessages.json';
 
+type TsErrorMessageDb = Record<string, { code: number }>;
+
+interface TSDiagnosticMatcher {
+  regexGlobal: RegExp;
+  regexLocal: RegExp;
+  parameters: string[];
+}
+
+const DiagnosticHashMap = new Map<string, TSDiagnosticMatcher>();
+
+const escapeRegex = /[.*+?^${}()|[\]\\]/g;
+
 function escapeRegExp(str: string) {
-  return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+  return str.replace(escapeRegex, '\\$&'); // $& means the whole matched string
 }
 
-let regexCache: Record<string, RegExp> = {};
+const parameterRegex = /{(\d)}/g;
```

Comment:
Moved the regexes outside of functions so they don't get recreated (surprisingly costly to recreate them).

Reply:
As long as they're global is safe to do so 👍